src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -490,9 +490,12 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_am_isend_eager(int rank, MPIR_Comm * c
 
         MPL_pointer_attr_t attr;
         MPIR_GPU_query_pointer_attr(buf, &attr);
-        if (attr.type == MPL_GPU_POINTER_DEV && !MPIDI_OFI_ENABLE_HMEM) {
-            /* Force packing of GPU buffer in host memory */
-            need_packing = true;
+        if (attr.type == MPL_GPU_POINTER_DEV) {
+            MPIDI_OFI_unregister_am_bufs();
```

Comment:
You mean `MPIDI_OFI_register_am_bufs()`?

Reply:
oops. will fix.